### PR TITLE
[GE] Grid UX Fixes and Minor Bug Fix for Property Grid

### PR DESCRIPTION
**Pull request by msDestiny14**
https://github.com/BabylonJS/Babylon.js/issues/11119
https://github.com/BabylonJS/ThePirateCove/issues/78

### CODE REVIEW

File: guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx [hunk 1]
```diff
@@ -101,7 +101,10 @@ export class TreeItemSelectableComponent extends React.Component<ITreeItemSelect
             return null;
         }
 
-        const children = Tools.SortAndFilter(entity, entity.getChildren ? entity.getChildren() : entity.children);
+        const grid = entity.typeName === "Grid" ;
+        console.log(grid);
```
Comment (sebavan) [inline guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx:105]: not sure we should keep this

File: guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx [hunk 2]
```diff
@@ -101,7 +101,10 @@ export class TreeItemSelectableComponent extends React.Component<ITreeItemSelect
             return null;
         }
 
-        const children = Tools.SortAndFilter(entity, entity.getChildren ? entity.getChildren() : entity.children);
+        const grid = entity.typeName === "Grid" ;
+        console.log(grid);
+        const children = grid ? Tools.SortAndFilterForGrid(entity, entity.getChildren ? entity.getChildren()
```
Comment (sebavan) [inline guiEditor/src/components/sceneExplorer/treeItemSelectableComponent.tsx:106]: This is kind of hard to read ?

File: guiEditor/src/tools.ts [hunk 3]
```diff
@@ -43,4 +43,27 @@ export class Tools {
         }
         return finalArray.reverse();
     }
+
+    public static SortAndFilterForGrid(parent: any, items: any[]): any[] {
```
Comment (sebavan) [inline guiEditor/src/tools.ts:47]: could we have a bit of typings ?

**Review by sebavan — COMMENTED** [review]
Only small comments, @RaananW will review more tomorrow.

**Comment by msDestiny14** [comment]
closing since there was some design changes

**Pull request closed without merging**